client/src/connection_cache.rs

```diff
@@ -241,20 +264,31 @@ fn get_or_add_connection(addr: &SocketAddr) -> GetConnectionResult {
             let mut map = (*CONNECTION_MAP).write().unwrap();
             get_connection_map_lock_measure.stop();
 
-            lock_timing = lock_timing.saturating_add(get_connection_map_lock_measure.as_ms());
+            lock_timing_ms = lock_timing_ms.saturating_add(get_connection_map_lock_measure.as_ms());
 
-            // evict a connection if the map is reaching upper bounds
-            while map.map.len() >= MAX_CONNECTIONS {
+            // evict a connection if the cache is reaching upper bounds
+            let mut num_evictions = 0;
+            let mut get_connection_cache_eviction_measure =
+                Measure::start("get_connection_cache_eviction_measure");
+            while map.list_of_peers.len() >= MAX_CONNECTIONS {
                 let mut rng = thread_rng();
                 let n = rng.gen_range(0, MAX_CONNECTIONS);
-                if let Some((nth_addr, _)) = map.map.iter().nth(n) {
-                    let nth_addr = *nth_addr;
-                    map.map.remove(&nth_addr);
-                }
+                let nth_addr = map.list_of_peers.remove(n);
```

Comment:
remove is O(n), maybe use swap_remove? See https://doc.rust-lang.org/std/vec/struct.Vec.html#method.remove

Reply:
Good point. I'll update the PR.